techniques/system/common/1.0/cron_setup.st

```diff
@@ -80,24 +87,17 @@ bundle agent check_cron_daemon
 
   methods:
 
-    pass2.!redhat.!fedora.restart_crond::
-
-      "restart_cron" usebundle => service_restart("cron");
-
-    pass2.(redhat|fedora).restart_crond::
-
-      "restart_cron" usebundle => service_restart("crond");
-
-    pass3.(service_restart_cron_not_ok|service_restart_crond_not_ok)::
-      "any" usebundle => rudder_common_report("Common", "result_error", "&TRACKINGKEY&", "CRON Daemon", "None", "The CRON daemon was not running and could not be restarted");
+    pass2.restart_crond::
+      "restart_cron" usebundle => service_restart("${service_name}");
 
-    pass3.(service_restart_cron_repaired|service_restart_crond_repaired)::
-      "any" usebundle => rudder_common_report("Common", "result_repaired", "&TRACKINGKEY&", "CRON Daemon", "None", "The CRON daemon has been successfully restarted");
+    # Fake success result, since we don't actually call service_restart unless it's needed
+    pass2.!restart_crond::
+      "any" usebundle => _classes_success("service_restart_${service_name}");
```

Comment:
Yes, exactly. So when it's *not* set we declare success. But actually, we should add in ```.!(android|windows|aix)``` to avoid duplicate report with the NA below!